common/src/main/java/io/druid/math/expr/Function.java

```diff
@@ -792,4 +874,172 @@ public ExprEval apply(List<Expr> args, Expr.ObjectBinding bindings)
       return eval.isNull() ? args.get(1).eval(bindings) : eval;
     }
   }
+
+  class ConcatFunc implements Function
+  {
+    @Override
+    public String name()
+    {
+      return "concat";
+    }
+
+    @Override
+    public ExprEval apply(List<Expr> args, Expr.ObjectBinding bindings)
+    {
+      if (args.size() == 0) {
+        return ExprEval.of(null);
+      } else {
+        final StringBuilder builder = new StringBuilder(Strings.nullToEmpty(args.get(0).eval(bindings).asString()));
```

Comment:
I thought it would be slightly more efficient, since it gives a slightly better hint to the StringBuilder about what the initial buffer size should be.